filesystem/src/main/java/org/frankframework/filesystem/AbstractFileSystemListener.java

```diff
@@ -400,25 +415,49 @@ public RawMessageWrapper<F> changeProcessState(RawMessageWrapper<F> message, Pro
 			}
 			if (toState==ProcessState.INPROCESS && isFileTimeSensitive() && getFileSystem() instanceof IWritableFileSystem) {
 				F movedFile = getFileSystem().moveFile(message.getRawMessage(), getStateFolder(toState), false);
-				String newName = getFileSystem().getCanonicalName(movedFile)+"-"+(DateFormatUtils.format(getFileSystem().getModificationTime(movedFile), DateFormatUtils.FULL_ISO_TIMESTAMP_NO_TZ_FORMATTER).replace(":", "_"));
-				F renamedFile = getFileSystem().toFile(newName);
-				int i=1;
-				while(getFileSystem().exists(renamedFile)) {
-					renamedFile=getFileSystem().toFile(newName+"-"+i);
-					if(i>5) {
-						log.warn("Cannot rename file [{}] with the timestamp suffix. File moved to [{}] folder with the original name", ()->message, ()->getStateFolder(toState));
-						return wrap(movedFile, message);
-					}
-					i++;
-				}
-				return wrap(FileSystemUtils.renameFile((IWritableFileSystem<F>) getFileSystem(), movedFile, renamedFile, false, 0), message);
+				return wrap(renameFileWithTimeStamp(message, toState, movedFile), message);
 			}
 			return wrap(getFileSystem().moveFile(message.getRawMessage(), getStateFolder(toState), false), message);
 		} catch (FileSystemException e) {
 			throw new ListenerException("Cannot change processState to ["+toState+"] for ["+getFileSystem().getName(message.getRawMessage())+"]", e);
 		}
 	}
 
+	@Nonnull
+	@SuppressWarnings("unchecked")
+	private F renameFileWithTimeStamp(RawMessageWrapper<F> message, ProcessState toState, F movedFile) throws FileSystemException {
+
+		String fileModificationDate = getFormatFileModificationDate(movedFile).replace(":", "_");
+
+		String fullName = getFileSystem().getName(movedFile);
+		if (fullName.contains(fileModificationDate)) {
+			return movedFile;
```

Comment:
If the timestamp is already contained in the file, then use the file as-is. It must have already gone through the in-process folder once before.

Do not attempt to check if a file with same name already exists -- it does, it's this file.